src/main/src/main/java/org/geoserver/util/NoExternalEntityResolver.java

```diff
@@ -35,7 +35,7 @@ public InputSource resolveEntity(String publicId, String systemId) throws SAXExc
         }
         
         // allow schema parsing for validation (jar or external only)
-        if (systemId != null && systemId.matches("(?i)(jar:file|http)[^?#;]*\\.xsd")) {
+        if (systemId != null && systemId.matches("(?i)(jar:file|http|vfs)[^?#;]*\\.xsd")) {
```

Comment:
Q: Why not http? 
A: The wild card only checks http (so https will still match!)

The change itself is straight forward, we could probably use a comment to explain what we are letting through (since this code is fairly annoying to read/understand)

````
// http(s) - external schema reference
// jar - internal schema reference (as constructed on tomcat)
// vfs - internal schema reference (as constructed on boss)
```